### DISCUSSION TITLE
Can't run examples after built

### DISCUSSION THREAD

**Discussion by lzt007** [faaxm/spix#53] [Q&A] [answered]
I have ran "install-deps.sh" in /ci  (I delete the --config in install-deps.sh )
Then I build spix with cmake-3.23.0-rc2,using the following command:
-----------------------------------
export CMAKE_PREFIX_PATH=/my/qt/path
mkdir build && cd build
cmake ..
cmake --build . --target install
------------------------------------
After that, I open the RemoteCtrl in qtcreator and try to debug it
Then I got a serious errors:
------------------------------------
>:-1: error: /usr/local/lib/x86_64-linux-gnu/libSpix.a(AnyRpcServer.cpp.o): in function `std::__cxx11::basic_string<char, >std::char_traits<char>, std::allocator<char> > spix::utils::unpackAnyRpcParam<std::__cxx11::basic_string<char, >std::char_traits<char>, std::allocator<char> > >(anyrpc::Value&)':
>AnyRpcServer.cpp:-1: error: undefined reference to `anyrpc::Value::GetString() const'
...
...
-------------------------------------
and many other errors like that says cannot find xxx.cpp in some other qt modules
I have installed both QT and AnyRpc,but it looks like I didn't confirm them well.
Is there a more more detailed description for the installation such as how to set the parameters,which ubuntu/qt/cmake version should I use?

And,as you can see, i'm a noob both in coding and English
I hope my question doesn't offend you

**Comment by faaxm** [comment, marked as answer]
Rest assured, I am not offended by your question, but took the liberty to move it over to the discussion page 😉
It looks like the build in qtcreator does not properly find AnyRPC and maybe some other dependencies.

Since you removed the `--config` option when building the dependencies, you probably only built the release version of them and are missing the debug version. When you then try to run the example in a debugger, qtcreator will try to link the example with the debug version of anyrpc, not find it and then show the errors you see.

The build types that are used by the CI can be found here:
https://github.com/faaxm/spix/blob/d260b51a4702a753455d1d677f4aabad15e1d281/.github/workflows/build.yml#L15
As you can see, there is `Release` and `Debug`. Try installing the dependencies again and this time, instead of removing `--config` run it with `--config Debug`. Also repeat the install steps for the dependencies.

There is also a pull request open with quite a few improvements to spix. Among them an update to the build documentation in the README which might be of help:
https://github.com/faaxm/spix/blob/4089b6f6f03f2d20a0ddd2856c527bb1a9964558/README.md

**Comment by lzt007** [comment]
Thanks for asking my question.I finally finished building the examples.
As I try the built in examples,everything works well. But when I try spix in my own project,I got some troubles in finding the items.
I made a simple  cmake qt project,with a button and a text in it.
I bound a signal to the button which will clear the text when it was clicked.
The items in .ui was set like the following :

widget class="QMainWindow" name="MainWindow"
==widget class="QWidget" name="centralwidget"
====widget class="QPushButton" name="pushButton"
====widget class="QTextEdit" name="textEdit"

I connect to 9000 port as the same in example,and try to click the button using the following python command.

 >s=xmlrpc.client.ServerProxy('http://localhost:9000')
>s.mouseClick("MainWindow/centralwidget/pushButton")

The connecting was success and the mouseClick func seems to be called correctly with a response ' '.But nothing was happen in app,the text in textEdit wasn't been cleared.
Then I try:

>s.existsAndVisible("MainWindow")

The response was False.It seems that spix can't find the MainWIndow as well as the items in it.
As the example was a qt-quick project.Maybe there are some differences when using spix in qt-widgets project?

What did  I do wrong?

**Reply by faaxm** [reply]
I haven't looked into running Spix with qt widgets, but there is an open issue to add support for it. The changes would require some cleanup to make it an optional feature for Spix (plus some added examples and tests) and I haven't had time to look into it yet.

See issue #8

**Reply by lzt007** [reply]
Thx.I got that
